course-v2/assets/js/video_download_popup.ts

```diff
@@ -0,0 +1,34 @@
+export const initVideoDownloadPopup = () => {
+  const downloadIcons = document.querySelectorAll(".video-download-icons")
+  const popups = document.querySelectorAll(".video-tab-download-popup")
+  let activePopup: HTMLElement | null = null
+
+  if (downloadIcons.length > 0) {
+    downloadIcons.forEach((downloadIcon, index) => {
+      const popup = popups[index] as HTMLElement
+      downloadIcon.addEventListener("click", event => {
+        event.stopPropagation()
```

Comment:
**Suggestion:** You can avoid keeping track of `activeElement` by restructuring this if you want.

Currently:
- (A) every download button has its own event listener
- (B) the overall document has an event listener

usually, (B) is responsible for closing the popup on "clicks outside" of the open popup. Except when the "click outside" ahppens to be another download button, then (A) is responsible.

A common way to deal with "click outside" behavior is to dynamically attach a closing listener on the opening click:
```ts
  downloadIcons.forEach((downloadIcon, index) => {
    const popup = popups[index]
    downloadIcon.addEventListener("click", event => {
      popup.classList.remove("hidden")
      const remove = (e: Event) => {
        if (event === e) return         // do not close popup when the original event bubbles up
        popup.classList.add("hidden")
        document.removeEventListener("click", remove) // cleanup after closing
      }
      document.addEventListener("click", remove)
    })
  })
```
Note that in this scenario you need to remove the "closing listener" once it has done its job.

Up to you.

Reply:
Sticking to the previous (current) code. Also, for sake of completeness, going to mention the complete modified code here:
```
export const initVideoDownloadPopup = () => {
  const downloadIcons = document.querySelectorAll(".video-download-icons")
  const popups = document.querySelectorAll(".video-tab-download-popup")

  downloadIcons.forEach((downloadIcon, index) => {
    const popup = popups[index]
    downloadIcon.addEventListener("click", event => {
      if (!popup.classList.contains("hidden")) return // Abort if popup is already open
      popup.classList.remove("hidden")
      const remove = (e: Event) => {
        if (event === e) return         // do not close popup when the original event bubbles up
        popup.classList.add("hidden")
        document.removeEventListener("click", remove) // cleanup after closing
      }
      document.addEventListener("click", remove)
    })
  })
}
```